src/mumble/LegacyPlugin.cpp

```diff
@@ -195,11 +194,26 @@ uint8_t LegacyPlugin::initPositionalData(const char *const *programNames, const
 		std::multimap< std::wstring, unsigned long long int > pidMap;
 
 		for (size_t i = 0; i < programCount; i++) {
-			std::string currentName = programNames[i];
-			std::wstring currentNameWstr =
-				std::wstring_convert< std::codecvt_utf8< wchar_t > >().from_bytes(currentName);
-
-			pidMap.insert(std::pair< std::wstring, unsigned long long int >(currentNameWstr, programPIDs[i]));
+			const std::string currentName = programNames[i];
+
+			try {
+				switch (sizeof(wchar_t)) {
```

Comment:
I have seen this implementation in at least two places. Why not turn it into a function and reuse it?

Reply:
I confirm there are two instances of it. Consider, though, that the stuff in `plugins` should be standalone (i.e. not depending directly on files from `src`).

Reply:
Alright - we should probably still create a dedicated function in both directories that can be reused by other code without having to rip it out of existing code.

Reply:
Keep in mind that Link is the only plugin that will ever need such a logic, for one precise reason: its API makes use of `wchar_t`, which is 2 bytes on Windows and 4 bytes on everything else.

Other plugins have to convert from the correct encoding based on the game they target.

Of course there may be cases where a game internally uses `wchar_t` and is built for both Linux and Windows. However, in such cases we cannot rely on `sizeof(wchar_t)` anyway because it would refer to the current process (Mumble client).